ngraph/core/include/openvino/core/layout.hpp

```diff
@@ -9,11 +9,20 @@
 
 #include "ngraph/attribute_adapter.hpp"
 #include "openvino/core/core_visibility.hpp"
+#include "openvino/core/partial_shape.hpp"
 #include "openvino/core/rank.hpp"
 #include "openvino/core/variant.hpp"
 
 namespace ov {
 
+class Layout;
+
+namespace layout {
+
+std::vector<int64_t> find_permutation(const Layout& src_layout, const Rank& src_shape_rank, const Layout& dst_layout);
```

Comment:
BTW, probably we dynamic rank / partial layout we would add more tests for this function specifically